Add age_in_millis to ILM Explain Response

gmjehovich opened the pull request:
This PR enhances the `_ilm/explain` API by adding an `age_in_millis` field to its response.

Previously, the index's "age" was only reported in a human-readable format (`age: "1.97d"`). 

This change introduces the `age_in_millis` field, providing the index's age as a precise millisecond value (e.g., `age_in_millis: 170208000`).

This new field is validated with corresponding unit and integration testing

Closes #103659

Commit:
Add age_in_millis to ILM Explain Response

Commit:
Fix: Declare age_in_millis in ILM Explain Response Parser

gmjehovich committed:
Merge branch 'main' into enhancement/add_ilm_explain_age_in_millis

gmjehovich committed:
Merge branch 'main' into enhancement/add_ilm_explain_age_in_millis

Commit:
Add unit tests for age_in_millis field in IndexLifecycleExplainResponse

gmjehovich committed:
Merge branch 'main' into enhancement/add_ilm_explain_age_in_millis

elasticsearchmachine commented:
Hi @gmjehovich, I've created a changelog YAML for you.

gmjehovich committed:
Update docs/changelog/128866.yaml

gmjehovich committed:
Update docs/changelog/128866.yaml

elasticsearchmachine commented:
Hi @gmjehovich, I've updated the changelog YAML for you.

elasticsearchmachine commented:
Pinging @elastic/es-data-management (Team:Data Management)

joegallo reviewed (approved):
Two tiny comments:

1.) There's a comment block ahead that notes that the array of arguments coming into the parsing constructor ignores a few fields -- since this one will be ignored, we should add it to that comment. Look for this section:

```
            // a[13] == "age"
            // a[20] == "time_since_index_creation"
            // <--- your new thing would go here
```

2.) You can DRY the `getAge(nowSupplier)` call and pull it out as a local `final var` -- I realize this is pracitically inconsequential, but it allows the world to know I was here. 😉 

Other than that, LGTM!

joegallo committed:
Merge branch 'main' into enhancement/add_ilm_explain_age_in_millis

joegallo commented:
For the record, I was really flummoxed by how we'd handle disparity between the writer and the reader with a synthetic field like the `age_in_millis`, but then of course it's just received by the constructor and ignored (which, you know, simple enough). But having a comment to indicate that you **know** you're ignoring it makes it a little clearer that this wasn't just an oversight.

gmjehovich committed:
Update IndexLifecycleExplainResponse.java

Fix comments; follow DRY for getAge(nowSupplier) call in IndexLifecycleExplainResponse

gmjehovich committed:
Merge branch 'main' into enhancement/add_ilm_explain_age_in_millis

gmjehovich committed:
Merge branch 'main' into enhancement/add_ilm_explain_age_in_millis
